### PR TITLE
Migrate to GitHub Actions 

### PR DESCRIPTION
**Pull request by modrake**
Per: https://github.com/hashicorp/GHA-migration/issues/57 , migrates ci from Circleci to GitHub Actions.

- Preserves circleci test functionality but removes test formatter, as it was not being utilized.
- Adds CODEOWNERS and Dependabot.
- Removes Circleci build files.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+
+updates:
+  - package-ecosystem: "github-actions"
```
Comment (shore) [inline .github/dependabot.yml:4]: We want to look for updates to go deps for this repo, don't we?
```suggestion
  - package-ecosystem: "gomod"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: hashicorp/mog/ci
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+    branches: ["main", "ghactions-migration"]
```
Comment (shore) [inline .github/workflows/ci.yml:6]: Is `ghactions-migration` supposed to be removed before merge?

**Review by shore — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+    branches: ["main"]
+    tags: ["*"]
+
+env:
+  TEST_RESULTS_DIR: /tmp/test-results
+
```
Comment (loshz) [inline .github/workflows/ci.yml:11]: I don't think we need this?
```suggestion
```

**Pull request merged**